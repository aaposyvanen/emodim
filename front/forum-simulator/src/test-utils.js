import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'

import createRootReducer from './reducer'

// This can be used instead of rtlRender to easily render connected (Redux) components.
// Can be especially useful when children of the tested component are connected.
// There might be a better way, (e.g. shallow rendering) but at least this works.
// Inspiration for this approach:
// https://redux.js.org/recipes/writing-tests#connected-components

function render(
    ui,
    {
        initialState,
        store = createStore(createRootReducer, initialState),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {
        return <Provider store={store}>{children}</Provider>
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render }
