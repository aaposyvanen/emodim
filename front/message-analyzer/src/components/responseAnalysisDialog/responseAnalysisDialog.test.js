import React from 'react';
import { render } from "../../test-utils";
import AnalysisReport from "./responseAnalysisDialog";

test("AnalysisReport renders correctly", () => {
    const { container } = render(<AnalysisReport />);
    expect(container).toMatchSnapshot();
});
