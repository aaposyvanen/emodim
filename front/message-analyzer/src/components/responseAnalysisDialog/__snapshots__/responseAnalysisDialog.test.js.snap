// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalysisReport renders correctly 1`] = `
<div>
  <div
    class="analysis-report"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Reply
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
